src/crypto.rs

```diff
@@ -99,7 +99,8 @@ impl TryFrom<(&[u8], COSEContentType)> for X509PublicKey {
                     .curve_name()
                     .ok_or(WebauthnError::OpenSSLErrorNoCurveName)?;
 
-                if ec_curve != nid::Nid::X9_62_PRIME256V1 {
+                if ec_curve != nid::Nid::X9_62_PRIME256V1 && ec_curve != nid::Nid::SECP384R1 {
```

Comment:
It's -7 in my vector, but interestingly it seems your vector doesn't have and alg key.

Reply:
I wonder if it's to do with the reading that it can be "any curve" but "recommends" SECP256R1 and the -7 just denotes the hashing. Which is why they selected SECP384R1 with sha256, and -7.

Reply:
Also my test vector isn't the attstmt, it's the full response, you'd need to pass it through the code to get that part out to see if alg is present.

Reply:
Yeah, I just ran the attestation object through a python script to inspect the data.

Reply:
I just did another capture and the algorithm is still there.

Reply:
It's not about the presence of alg or not, it's if the value is -7 or -35, and what that means for accepting different curves in this function. There is a recommendation in the standard that says you *should* use SECP256R1 but it doesn't say you must. It makes me question if apples use of the crypto here is valid (because of the should statement) vs invalid (because it goes against the recommendation). It's not your implementation here that's at fault or at question, I'm wondering about the correctness of the data that apples devices are emitting.

Reply:
I just did a bit more sleuthing and there's some mention of the `alg` key in a dump someone posts in this PR:
https://github.com/web-auth/webauthn-framework/issues/148. I think it may be from an earlier version of Apple's webauthn format spec and later got dropped. I'm curious, however, as to why my phone continues to send attestation objects with it. For reference I'm on iOS 14.2.

Reply:
well part of this work is done by the secure enclave, so it could be the model of device in question? What hardware do you have? My challenge was on an ipad mini 5th gen, but I have *not* tried my iphone 7. So I wonder if that's related?

Reply:
> It's not about the presence of alg or not, it's if the value is -7 or -35, and what that means for accepting different curves in this function. There is a recommendation in the standard that says you _should_ use SECP256R1 but it doesn't say you must. It makes me question if apples use of the crypto here is valid (because of the should statement) vs invalid (because it goes against the recommendation). It's not your implementation here that's at fault or at question, I'm wondering about the correctness of the data that apples devices are emitting.

Right, I understand. I was just trying to confirm that my observation was correct before we went further down the rabbithole. I can confirm that my alg is -7.

> well part of this work is done by the secure enclave, so it could be the model of device in question? What hardware do you have? My challenge was on an ipad mini 5th gen, but I have _not_ tried my iphone 7. So I wonder if that's related?

I'm on an iPhone 8.

Reply:
See conversation below---this field has been removed from newer versions of webkit (although it's strange I'm still seeing it).